tests/storage_cases/test_seek_region.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use raftstore::server::new_server_cluster;
+use std::thread;
+use std::time::Duration;
+use tikv::raftstore::store::SeekRegionResult;
+use tikv::storage::engine::RegionInfoSource;
+use tikv::util::HandyRwLock;
+
+#[test]
+fn test_seek_region() {
+    // Prepare
+    let mut cluster = new_server_cluster(0, 3);
+    cluster.run();
+
+    for i in 0..15 {
+        let i = i + b'0';
+        let key = vec![b'k', i];
+        let value = vec![b'v', i];
+        cluster.must_put(&key, &value);
+    }
+
+    let end_keys = vec![
+        b"k1".to_vec(),
+        b"k3".to_vec(),
+        b"k5".to_vec(),
+        b"k7".to_vec(),
+        b"k9".to_vec(),
+        b"".to_vec(),
+    ];
+
+    let mut regions = Vec::new();
+
+    for mut key in end_keys.iter().take(end_keys.len() - 1).map(Vec::clone) {
+        let region = cluster.get_region(&key);
+        cluster.must_split(&region, &key);
+
+        key[1] -= 1;
+        let region = cluster.get_region(&key);
+        regions.push(region);
+    }
+    regions.push(cluster.get_region(b"k9"));
```

Comment:
could you add comments or tests to assert the region range?

Reply:
Done